File: CarePackage/Software/Insomnia.cs
namespace CarePackage.Software;

public class Insomnia : BaseSoftware
{
    public override string           Key            { get; set; } = "insomnia";
    public override string           Name           { get; set; } = "Insomnia";
    public override SoftwareCategory Category       { get; set; } = SoftwareCategory.Development;
    public override string           DownloadName   { get; set; } = "Insomnia.Core.exe";
    public override bool             IsArchive      { get; set; } = false;
    public override bool             ShouldCacheUrl { get; set; } = false;
    public override bool             RequiresAdmin  { get; set; } = false;
    public override Bitmap           Icon           { get; set; } = Resources.Icons.insomnia;
    public override string           Homepage       { get; set; } = "https://insomnia.rest";

    public override Task<string> GetDownloadUrlAsync(CancellationToken ct)
        => Task.FromResult("https://updates.insomnia.rest/downloads/windows/latest?app=com.insomnia.app&source=website");
}

File: packages/app/src/services/software/definitions/insomnia.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'insomnia';
	public name = 'Insomnia';
	public category = [
		SoftwareCategory.Development,
		SoftwareCategory.Network,
	];
	public downloadName = 'Insomnia.Core.exe';
	public isArchive = false;
	public shouldCacheUrl = false;
	public requiresAdmin = false;
	public deprecated = false;
	public icon = 'insomnia.png';
	public homepage = 'https://insomnia.rest';

	public async resolveDownloadUrl() {
		return ok('https://updates.insomnia.rest/downloads/windows/latest?app=com.insomnia.app&source=website');
	}
}
